### ISSUE TITLE
Conditional search space

### ISSUE THREAD

**Issue by MotoRZR** [scikit-optimize/scikit-optimize#700]
Is it possible to set up a conditional search space?  A good example would be if I'm using  a particular loss function for a tree based model I want the search space to include a particular hyperparameter but if I'm not using that loss function then I want to exclude that hyperparameter or restrict it to a certain range?

Thanks.
